app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -174,51 +153,66 @@
-        });
+      var deregisterNotificationListener = function() {
+        notificationListener && notificationListener();
+        notificationListener = null;
       };
 
-      // TODO: follow-on PR to decide which of these events to toast,
-      // via config in constants.js
       var eventWatchCallback = function(eventData) {
-        eventsByNameData = eventData.by('metadata.name');
-        eventsMap = formatAndFilterEvents(eventsByNameData);
-        // TODO: Update to an intermediate map, so that we can then combine both
-        // events + notifications into the final notificationGroups output
-        notificationGroups = sortNotificationGroups(eventsMap);
+        eventsMap[$routeParams.project] = formatAndFilterEvents(eventData.by('metadata.name'));
         render();
       };
 
-      // TODO: Follow-on PR to update & add the internal notifications to the
-      // var notificationWatchCallback = function(event, notification) {
-      //   // will need to add .event = {} and immitate structure
-      //   if(!notification.lastTimestamp) {
-      //     // creates a timestamp that matches event format: 2017-08-09T19:55:35Z
-      //     notification.lastTimestamp = moment.parseZone(new Date()).utc().format();
-      //   }
-      //   clientGeneratedNotifications.push(notification);
-      // };
+      var notificationWatchCallback = function(event, notification) {
+        var uid = notification.id || _.uniqueId('notification-');
+        var project = notification.namespace || $routeParams.project;
+        notificationsMap[project] = notificationsMap[project] || {};
+        notificationsMap[project][uid] = {
+          actions: null,
+          unread: true,
+          uid: uid,
+          // emulating an API event to simplify the template
+          event: {
+            lastTimestamp: notification.lastTimestamp || moment.parseZone(new Date()).utc().format(),
+            message: notification.message,
+            namespace: project,
+            type: notificationEventTypeMap[notification.type] || 'Normal',
+            links: notification.links
+          }
+        };
+        render();
+      };
 
-      var iconClassByEventSeverity = {
-        Normal: 'pficon pficon-info',
-        Warning: 'pficon pficon-warning-triangle-o'
+      var watchEvents = function(projectName, cb) {
+        deregisterEventsWatch();
+        if(projectName) {
+          eventsWatcher = DataService.watch('events', {namespace: projectName}, _.debounce(cb, 400), { skipDigest: true });
+        }
+      };
+
+      // NotificationsService notifications are minimal, they do no necessarily contain projectName info.
+      // ATM tacking this on via watching the current project.
+      var watchNotifications = function(projectName, cb) {
+        deregisterNotificationListener();
+        if(!projectName) {
+          return;
+        }
+        notificationListener = $rootScope.$on('NotificationsService.onNotificationAdded', cb);
```

Comment:
We should be able to add this once instead of registering and removing on project changes.

Reply:
True. no need to update this one like the events watch.